e3sm_to_cmip/__main__.py

```diff
@@ -97,9 +99,14 @@ class CLIArguments:
 
 class E3SMtoCMIP:
     def __init__(self, args: Optional[List[str]] = None):
+        global logger
```

Comment:
Added the commentary.

Since the logger assignment was moved inside of the class __init__ function, and specifically AFTER  the call to self._parse_args, the logger would not be available to other functions within the class if not made global.